runtime/near-vm-runner/src/tests/fuzzers.rs

```diff
@@ -167,3 +170,31 @@ fn wasmer2_and_wasmtime_agree() {
         assert_eq!(wasmer2, wasmtime);
     });
 }
+
+#[test]
+fn wasmer2_is_reproducible() {
+    bolero::check!().for_each(|data: &[u8]| {
+        if let Ok(module) = ArbitraryModule::arbitrary(&mut arbitrary::Unstructured::new(data)) {
+            let code = ContractCode::new(module.0.module.to_bytes(), None);
+            let config = VMConfig::test();
+            let mut first_hash = None;
+            for _ in 0..3 {
+                let vm = Wasmer2VM::new(config.clone());
+                let prepared_code = match crate::prepare::prepare_contract(code.code(), &config) {
+                    Ok(c) => c,
+                    Err(_) => return,
+                };
+                let exec = match vm.compile_uncached(&prepared_code) {
+                    Ok(e) => e,
+                    Err(_) => return,
+                };
+                let code = exec.serialize().unwrap();
+                let hash = CryptoHash::hash_bytes(&code);
+                match first_hash {
+                    None => first_hash = Some(hash),
+                    Some(h) => assert_eq!(h, hash),
+                }
```

Comment:
Maybe:

```
let first_hash = *first_hash.get_or_insert(hash);
assert_eq!(first_hash, hash);
```

Reply:
Hmm so this would definitely work, but I’m not sure the loss in readability is worth the gain in conciseness?

Reply:
What about the following pattern?

```
let mut prev_hash = None;
for 0..3 {
    ...
    if let Some(h) = prev_hash {
        assert_eq!(h, hash);
    }
    prev_hash = Some(hash);
}
```

Reply:
Or maybe a mix of the two, like this?
```
let mut first_hash = None;
for _ in 0..3 {
    // ...
    if let None = first_hash {
        first_hash = Some(hash);
    }
    assert_eq!(first_hash, Some(hash));
}
```

Now I must say I don’t have any strong preferences, so unless someone comes forward with a strong preference for one of these styles I’ll just go forward and land it :)

Reply:
I also don't have a strong opinion at all. I just saw a bike shed like discussion and felt like jumping in =).